eng/common/templates/steps/source-build.yml

```diff
@@ -37,10 +37,10 @@ steps:
     ${{ coalesce(parameters.platform.buildScript, './build.sh') }} --ci \
       --configuration $buildConfig \
       --restore --build --pack --publish -bl \
+      --sourceBuild \
       $officialBuildArgs \
       $targetRidArgs \
-      /p:SourceBuildNonPortable=${{ parameters.platform.nonPortable }} \
-      /p:ArcadeBuildFromSource=true
+      /p:SourceBuildNonPortable=${{ parameters.platform.nonPortable }}
```

Comment:
⚠️  Changes in this file will need to be ported to dotnet/arcade or they will break during the next update

Reply:
Options here:

* Merge into feature branch with this form, but update arcade and incorporate here before merging back to main
* Update **eng/build.sh** to detect `/p:ArcadeBuildFromSource=true` instead of `--sourceBuild` (how?) and revert this change

Reply:
We should do number 2.

Reply:
I might be naive, but I believe it should be possible to look through `$properties` and find a property with that string.

Reply:
I believe this should now be fixed.

Reply:
Actually, @MichaelSimons does this flag need to passed all the way through our CI script? The pipeline passed, but I see a number of conditions in arcade based on that flag.